waspc/src/Wasp/Message.hs

```diff
@@ -0,0 +1,5 @@
+module Wasp.Message (MessageType (..), SendMessage) where
+
+data MessageType = Start String | Success String | Warning String | Error String
+
+type SendMessage = MessageType -> IO ()
```

Comment:
I hope after the discussion about this yesterday and Filip's points you've changed your mind on this. This would force us to do:

```haskell
setup :: HandleMessage -> Something
setup handleMessage = handleMessage $ SuccessMessage "Hello" 
```

which I think is ugly, because we're not handling any messages here, we just want to send it.

Or this:

```haskell
setup :: HandleMessage -> Something
setup sendMessage = sendMessage $ SuccessMessage "Hello"
```

where we have a different type name and argument name.

I think in general when you're injecting dependencies you want to think of the perspective of who is going to use it, not that someone is going to handle this. After all it's only a function you get through an argument instead of by referring to it in the same module or importing it from somewhere else.

Reply:
Hey, agree on this. Second code example is certainly not nice. First example is ok for both of us I believe, the only question is if correct verbe is "handle" or "send".

So let's look at that: "handle" means give it to somebody do something with it. "send" means send it to somebody. Very similar. I am actually not sure what is the difference. I think I liked "handle" better because it kind of gives the responsibility of what to do with a message to the handler. So I am handing off the message, and the handler will decide what to do with it. Which I think is the case here.
On the other hand, "send" doesn't presume much -> it says I am sending this message to someone. So more focus on the act of sending, of the message traveling. Nothing wrong with that.

So to conclude, they are so similar that I think both are ok and am ok with any option. But just for curiosity, would love to hear a bit on why you think "send" makes more sense than "handle".